tokio/src/sync/mpsc/unbounded.rs

```diff
@@ -172,6 +172,56 @@ impl<T> UnboundedReceiver<T> {
         poll_fn(|cx| self.poll_recv(cx)).await
     }
 
+    /// Receives the next values for this receiver and extends `buffer`.
+    ///
+    /// This method returns the number of values populated in `buffer`.
+    ///
+    /// This method returns 0 if the channel has been closed and there are
+    /// no remaining messages in the channel's queue. This indicates that no
+    /// further values can ever be received from this `Receiver`. The channel is
+    /// closed when all senders have been dropped, or when [`close`] is called.
+    ///
+    /// If there are no messages in the channel's queue, but the channel has
+    /// not yet been closed, this method will sleep until a message is sent or
+    /// the channel is closed.
```

Comment:
I'd like to add some more emphasis on the fact that if it returns 0, then the channel is closed. Perhaps we can swap these paragraphs, and add something to the paragraph about 0 along the lines of "this method will never return 0 if the channel is not closed" or "if the channel is not closed, then this method will never return 0".

Reply:
The zero case is analogous to the `None` case with `recv`.  Does it make sense to change `recv`'s documentation as well?

I think the simplest accommodation is to add _only_, e.g.,:
_This method returns 0 only if..._
_This method returns `None` only if..._

Reply:
I don't think it's quite as important for `recv` because people are less likely to be confused about `None`. I've seen confusion about 0 = closed a lot of times with the `read` method from the IO traits.

Reply:
I've again strengthened the language in today's push,

> This method will never return 0 unless the channel has been closed and
> there are no remaining messages in the channel's queue.

Does this work?

Reply:
Yes, it looks great now.